lib/server.js

```diff
@@ -489,6 +490,61 @@ Server.prototype.pre = function pre() {
     return this;
 };
 
+// eslint-disable-next-line jsdoc/check-param-names
+/**
+ * Gives you hooks that run before restify touches a request. These hooks
+ * allow you to do processing early in the request/response life cycle without
+ * the overhead of the restify framework. You can not yield the event loop in
+ * this handler.
+ *
+ * This is useful in cases where you want to make quick decisions about a
+ * request, for example when shedding load.
+ *
+ * The function handler accepts two parameters: req, res. If you want restify
+ * to ignore this request, return false from your handler. Return true or
+ * undefined to let restify continue handling the request.
+ *
+ * When false is returned, restify immediately stops handling the request. This
+ * means that no further middleware will be executed for any chain and routing
+ * will not occure. All request/response handling for an incoming request must
+ * be done inside the first handler if you intend to return false. This
+ * includes things like closing the response and returning a status code.
+ *
+ * The only work restify does for a first handler is to increment the number of
+ * inflightRequests prior to calling the chain, and decrement that value if the
+ * handler returns false.
+ *
+ * @public
+ * @memberof Server
+ * @instance
+ * @function first
+ * @param {...Function} handler - Allows you to add handlers that
+ * run for all requests, *before* restify touches the request.
+ * This gives you a hook to change request headers and the like if you need to.
+ * Note that `req.params` will be undefined, as that's filled in *after*
+ * routing.
+
+ * Takes one or more functions.
+ * @returns {Object} returns self
+ * @example
+ * sever.first(function(req, res) {
+ *   if(server._inflightRequests > 100) {
```

Comment:
Should the sample code be using private vars? Any reason not to use the public `inflightRequests()` API?